QModManager/Utility/MainMenuMessages.cs

```diff
@@ -0,0 +1,169 @@
+﻿namespace QModManager.Utility
+{
+    using System.Reflection;
+    using System.Reflection.Emit;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    using Harmony;
+    using UnityEngine;
+    using UnityEngine.SceneManagement;
+
+    /// <summary>
+    /// Allows to add critical messages to the main menu.
+    /// Messages will stay in the main menu and on loading screen.
+    /// </summary>
+    public static class MainMenuMessages
+    {
+        private const int defaultSize = 25;
+        private const string defaultColor = "red";
+
+        private static List<string> messageQueue;
+        private static List<ErrorMessage._Message> messages;
+
+        /// <summary> Adds message to the main menu. </summary>
+        /// <param name="msg"> message to add </param>
+        /// <param name="size"> text size </param>
+        /// <param name="color"> text color </param>
+        /// <param name="autoformat"> whether it needed to apply formatting tags to the message or show it as it is </param>
+        public static void Add(string msg, int size = defaultSize, string color = defaultColor, bool autoformat = true)
+        {
+            if (Patches.hInstance == null) // just in case
+            {
+                Logger.Error($"MainMenuMessages: trying to add message before Harmony initialized ({msg})");
```

Comment:
You can remove the `MainMenuMessages:` prefix, since that is already added by the `Logger` class